### ISSUE TITLE
Labeling buttons to improve accessibility for the blind

### ISSUE THREAD

**Issue by MahfudNay** [signalapp/Signal-iOS#864]
Hello

Congratulation for Signal app, it is a fantastic app, now it is the only app that blind users can use for private messages.
To improve the accessibility, you have several tools from Apple here https://developer.apple.com/accessibility/
I would like to suggest some changes to improve accessibility for blind users for Signal app.
I can press buttons but some of them are not correctly labeled, I know what happen because I've saw the result of the action but VoiceOver reads something different that I do not understand.
1. Delete a conversation: when I put down a finger to see what actions i can get, VoiceOver does not read anything when I'm on the delete  option and the app do not ask me before if I'm sure to want to delete the conversation, so, I've deleted when I didn't want to do that.
2. Attach something: Voiceover reads "BtnAttachment Blue". I recommend you to label it as "Attach something, Attach or share".
1. BtnCancel white: I recommend to label it as "Cancel or back". That happen for example when I check the settings of the app.
   1. BtnPhone white: in an individual conversation, VoiceOver reads that button to call someone. I recommend to label it as "Phone, call or phone (name of the person) or call to (name of the person).
   2. Separated  information for a message: when I receive a message, I have to flick 3 times to read all information about the message: A. Time. B. Content of the message. C. Name of the person. 
      I recommend you to try to get that VoiceOver reads all that information just in once.
   3. The name of the person doesn't appear for each message in a conversation. When I read the content of the conversation, I have to guess if the message that I am reading is mine or from the my friend. If you get that VoiceOver say the name of the person before read the content of the message, then, it would be perfect :).

**Comment by FredericJacobs** [comment]
Oh, wow. Thanks for writing this up.

I'll be in touch with you soon. I would love you to be part of our beta-program so that we can have shorter feedback loop (so that we don't have to wait for Apple to approve the test updates).

**Comment by MahfudNay** [comment]
Hello

Thank you very much.
I will be very happy to  participate in the beta program, I wait  for your instructions.
I’m sure that  working together we are going to get a good app :).

Mahfud




Mahfud Chaaban
mahfud.chaaban@gmail.com
www.tecnocharlas.es <http://www.tecnocharlas.es/>

**Comment by riyapenn** [comment]
Additional voiceover details:
"Voiceover does not properly identify the contact's associated icon/control as being greyed out. To that end, attempting to search for a person by name/number doesn't readily give me any feedback either; instead prospective Voiceover users get stuck in the text entry field with no other obvious controls except for "cancel" and "clear text" buttons."

**Comment by MahfudNay** [comment]
Hello
 I haven’t received any new news about updates including accessibility.
How is the accessibility going on Signal? 
Last week I update Signal but I didn’t see anything new about accessibility, for me it is exactly the same app.
I hope you label and include new features in the next update :).
Thank you for your work.
Mahfud

**Comment by MahfudNay** [comment]
Hello 

Have you forogot the accessibility for the blind  users on iPhone?
 I am sad, I sent a report in Agust and you promise to me that you are going to  improve it and I would send me a beta to tester it. You  released two updates after that  and any of them has accessibility improvement.
I love this aplicación and I use it everyday almost, but the accessibility should improve please.

Kind regards

Mahfud Chaaban
mahfud.chaaban@gmail.com
www.tecnocharlas.es

**Comment by MarcoZehe** [comment]
Related to #1210.
